File: android/src/main/java/com/helloworldplugin/HelloWorld.java
package com.helloworldplugin;

import android.util.Log;

public class HelloWorld {

    public String echo(String value) {
        Log.i("HelloWorld", "Echo called with: " + value);
        return value;
    }

    public String getGreeting(String name) {
        String greeting = "Hello " + name + " from Android!";
        Log.i("HelloWorld", "Greeting: " + greeting);
        return greeting;
    }
}

File: ios/Plugin/HelloWorld.swift
import Foundation

@objc public class HelloWorld: NSObject {
    @objc public func echo(_ value: String) -> String {
        print("Hello World Plugin - Echo: \(value)")
        return value
    }

    @objc public func getGreeting(_ name: String) -> String {
        let greeting = "Hello \(name) from iOS!"
        print("Hello World Plugin - Greeting: \(greeting)")
        return greeting
    }
}